sdk/webpubsub/azure-resourcemanager-webpubsub/src/test/java/com/azure/resourcemanager/webpubsub/WebPubSubManagerTests.java

```diff
@@ -0,0 +1,121 @@
+import java.util.Collections;
+import java.util.Random;
+
+public class WebPubSubManagerTests extends TestBase {
+    private static final Random RANDOM = new Random();
+    private static final Region REGION = Region.US_EAST2;
+    private String resourceGroupName = "rg" + randomPadding();
+    private WebPubSubManager webPubSubManager;
+    private ResourceManager resourceManager;
+    private boolean testEnv;
+
+    @Override
+    public void beforeTest() {
+        final TokenCredential credential = new DefaultAzureCredentialBuilder().build();
+        final AzureProfile profile = new AzureProfile(AzureEnvironment.AZURE);
+
+        webPubSubManager = WebPubSubManager
+            .configure()
+            .withLogOptions(new HttpLogOptions().setLogLevel(HttpLogDetailLevel.BASIC))
+            .authenticate(credential, profile);
+
+        resourceManager = ResourceManager
+            .configure()
+            .withLogOptions(new HttpLogOptions().setLogLevel(HttpLogDetailLevel.BASIC))
+            .authenticate(credential, profile)
+            .withDefaultSubscription();
+
+        // use AZURE_RESOURCE_GROUP_NAME if run in LIVE CI
+        String testResourceGroup = Configuration.getGlobalConfiguration().get("AZURE_RESOURCE_GROUP_NAME");
+        testEnv = !CoreUtils.isNullOrEmpty(testResourceGroup);
+        if (testEnv) {
+            resourceGroupName = testResourceGroup;
+        } else {
+            resourceManager.resourceGroups()
+                .define(resourceGroupName)
+                .withRegion(REGION)
+                .create();
+        }
+    }
+
+    @Override
+    protected void afterTest() {
+        if (!testEnv) {
+            resourceManager.resourceGroups().beginDeleteByName(resourceGroupName);
+        }
+    }
+
+    @Test
+    @DoNotRecord(skipInPlayback = true)
+    public void testCreateWebPubSubResource() {
+        WebPubSubResource webPubSubResource = null;
+        try {
+            String resourceName = "webpubsub" + randomPadding();
+            // @embedmeStart
+            webPubSubResource = webPubSubManager.webPubSubs()
+                .define(resourceName)
+                .withRegion(REGION)
+                .withExistingResourceGroup(resourceGroupName)
+                .withSku(new ResourceSku().withName("Premium_P1").withCapacity(1).withTier(WebPubSubSkuTier.PREMIUM))
+                .withTls(new WebPubSubTlsSettings().withClientCertEnabled(false))
+                .withNetworkACLs(new WebPubSubNetworkACLs()
+                    .withDefaultAction(AclAction.DENY)
+                    .withPublicNetwork(new NetworkAcl().withAllow(
+                        Arrays.asList(WebPubSubRequestType.SERVER_CONNECTION,
+                            WebPubSubRequestType.CLIENT_CONNECTION,
+                            WebPubSubRequestType.RESTAPI,
+                            WebPubSubRequestType.TRACE)
+                    ))
+                    .withPrivateEndpoints(Collections.emptyList())
```

Comment:
remove if not needed

Reply:
Redundant parameter settings have been removed and `Sku` type has been changed to `Free` in the new version.